presto-spark-base/src/main/java/com/facebook/presto/spark/execution/LocalTask.java

```diff
@@ -0,0 +1,261 @@
+import com.facebook.presto.execution.buffer.OutputBuffers;
+import com.facebook.presto.execution.buffer.PageBufferInfo;
+import com.facebook.presto.execution.scheduler.TableWriteInfo;
+import com.facebook.presto.operator.TaskStats;
+import com.facebook.presto.server.SimpleHttpResponseCallback;
+import com.facebook.presto.server.SimpleHttpResponseHandler;
+import com.facebook.presto.server.TaskUpdateRequest;
+import com.facebook.presto.server.remotetask.RemoteTaskStats;
+import com.facebook.presto.server.smile.BaseResponse;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import org.joda.time.DateTime;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static com.facebook.airlift.http.client.Request.Builder.prepareGet;
+import static com.facebook.airlift.http.client.Request.Builder.preparePost;
+import static com.facebook.airlift.http.client.StaticBodyGenerator.createStaticBodyGenerator;
+import static com.facebook.presto.execution.TaskInfo.createInitialTask;
+import static com.facebook.presto.server.RequestHelpers.setContentTypeHeaders;
+import static com.facebook.presto.server.smile.AdaptingJsonResponseHandler.createAdaptingJsonResponseHandler;
+import static com.facebook.presto.spi.StandardErrorCode.LOCAL_TASK_ERROR;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class LocalTask
+{
+    private static final Logger log = Logger.get(LocalTask.class);
+    private final Session session;
+    private final TaskId taskId;
+    private final PlanFragment planFragment;
+    @GuardedBy("this")
+    private final AtomicReference<OutputBuffers> outputBuffers = new AtomicReference<>();
+    private final HttpClient httpClient;
+    private final TableWriteInfo tableWriteInfo;
+    private final RemoteTaskStats stats;
+    private final List<TaskSource> sources;
+    private TaskInfo taskInfo;
+    private final Executor executor;
+    private final ScheduledExecutorService updateScheduledExecutor;
+    @GuardedBy("this")
+    private ScheduledFuture<?> scheduledFuture;
+    private final JsonCodec<TaskUpdateRequest> taskUpdateRequestCodec;
+    private final JsonCodec<PlanFragment> planFragmentCodec;
+    private final JsonCodec<TaskInfo> taskInfoCodec;
+
+    public LocalTask(
+            Session session,
+            TaskId taskId,
+            String nodeId,
```

Comment:
Do we need this?

Reply:
Also, it would be good to sanity check other fields. Not all of them are needed for local execution.....

Reply:
Currently it's used when create Initial TaskInfo. I assume I can use a dummy node id in there?
BTW what will be the node id in the TaskInfo from presto_cpp side? will it be some actual node id or dummy ones